src/Orleans.Runtime/Utilities/SiloUnobservedExceptionsHandler.cs

```diff
@@ -3,16 +3,27 @@
 using Microsoft.Extensions.Logging;
 using Orleans.Runtime.Scheduler;
 using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.DependencyInjection.Extensions;
+using Orleans.Hosting;
 
 namespace Orleans.Runtime
 {
-    internal static class SiloUnobservedExceptionsHandlerServiceProviderExtensions
+    public static class SiloUnobservedExceptionsHandlerServiceProviderExtensions
     {
-        public static void InitializeSiloUnobservedExceptionsHandler(this IServiceProvider services)
+        internal static void InitializeSiloUnobservedExceptionsHandler(this IServiceProvider services)
         {
             //resolve handler from DI to initialize it
             var ignore = services.GetService<SiloUnobservedExceptionsHandler>();
```

Comment:
`InitializeSiloUnobservedExceptionsHandler` is still called in the `Silo` ctor, no? So it should fail if the user didn't registered the `SiloUnobservedExceptionsHandler`, or am I missing something?

Reply:
no it won't fail because it is just try to get UnObservedHandler from DI. if there's not one then it returns null, which is fine

Reply:
Ah yes, you're right, it returns null here